src/toplevel.c

```diff
@@ -791,7 +794,11 @@ jl_value_t *jl_toplevel_eval_flex(jl_module_t *JL_NONNULL m, jl_value_t *e, int
     thk = (jl_code_info_t*)jl_exprarg(ex, 0);
     assert(jl_is_code_info(thk));
```

Comment:
I think it would be enough to turn this assert into an error.

Reply:
Changed the assert to error in a7c56d7